robot/impl/local_robot.go

```diff
@@ -178,13 +178,23 @@ func (r *localRobot) StopAll(ctx context.Context, extra map[resource.Name]map[st
 	return nil
 }
 
-// Config returns the config used to construct the robot. Only local resources are returned.
-// This is allowed to be partial or empty.
-func (r *localRobot) Config(ctx context.Context) (*config.Config, error) {
-	cfgCpy := *r.config
-	cfgCpy.Components = append([]resource.Config{}, cfgCpy.Components...)
-
-	return &cfgCpy, nil
+// Config returns a config representing the current state of the robot.
+func (r *localRobot) Config() *config.Config {
+	cfg := r.mostRecentCfg
+
+	// Use resource manager to generate Modules, Remotes, Components, Processes
+	// and Services.
+	//
+	// NOTE(benji): it would be great if the resource manager could somehow
+	// generate Cloud, Packages, Network and Auth fields.
```

Comment:
I don't think the resource manager can single-handedly generate the whole robot config. The `Cloud`, `Packages`, `Network` and `Auth` fields are all outside of the resource manager's current domain, and are all significant fields for reconfiguration. We may have to store a config and generate only part of it for the time being. I still think this approach is better than only storing a config, as we treat the resource manager/graph as the source of truth for the `Modules`, `Remotes`, `Components`, `Processes` and `Services` fields.